servicetalk-data-protobuf/src/main/java/io/servicetalk/data/protobuf/ProtobufSerializerFactory.java

```diff
@@ -109,8 +112,20 @@ private <T extends MessageLite> Parser<T> newParser(Class<T> clazz) {
         final MethodHandle mh;
         try {
             mh = MethodHandles.publicLookup().findStatic(clazz, PARSER_METHOD_NAME, PARSER_METHOD_TYPE);
-        } catch (IllegalAccessException | NoSuchMethodException e) {
+        } catch (IllegalAccessException e) {
             throw new IllegalArgumentException("Unable to find " + clazz + "." + PARSER_METHOD_NAME, e);
+        } catch (NoSuchMethodException nsme) {
+            // perhaps the class is a protobuf 2.x generated class
+            try {
```

Comment:
can we add this code to an independent method so it in the expected case (protbuf 3) the byte code for this method is smaller?